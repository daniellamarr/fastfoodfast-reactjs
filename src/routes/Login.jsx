import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userLogin } from '../actions/authActions';
import Loader from '../components/Loader';

class Login extends Component {
  constructor() {
    super();

    this.state = {
      email: '',
      password: '',
    }

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit(e) {
    e.preventDefault();

    this.props.userLogin({
      email: this.state.email,
      password: this.state.password,
    });
  }
  render() {
    return (
      <div id="loginPage">
        { this.props.auth.loadingStatus ? <Loader /> : null }
        <section id="login">
          <center>
            <div className="login">
              <h4>I see you are hungry once again, Please sign in</h4>
              <form id="loginForm" method="post" onSubmit={this.onSubmit}>
                <input type="email" name="email" id="email" placeholder="Email" onChange={this.onChange} required />
                <input type="password" name="password" id="password" placeholder="Password" onChange={this.onChange} required />
                <button type="submit">Sign in </button>
              </form>
            </div>
          </center>
        </section>
        <div className="error hide"></div>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { userLogin })(Login);
