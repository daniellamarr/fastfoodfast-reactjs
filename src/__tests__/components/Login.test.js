/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { mount } from 'enzyme';
import { Login } from '../../routes/Login.jsx';

let wrapper;

const props = {
  auth: {
    loadingStatus: false,
  },
  userLogin: () => {},
};

beforeEach(() => {
  wrapper = mount(<Login {...props} />);
});

describe('Login Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should find onChange()', () => {
    wrapper.find('#email').simulate('change');
    expect(wrapper.state().email).toEqual('');
  });

  it('should find onSubmit()', () => {
    wrapper.find('#loginForm').simulate('submit', {
      preventDefault: () => {},
    });
    const props2 = { ...props, auth: { loadingStatus: true } };
    wrapper = mount(<Login {...props2} />);
    expect(wrapper.props().auth.loadingStatus).toEqual(true);
  });
});
