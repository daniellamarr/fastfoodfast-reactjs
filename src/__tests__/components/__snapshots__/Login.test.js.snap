// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<Login
  auth={
    Object {
      "loadingStatus": false,
    }
  }
  userLogin={[Function]}
>
  <div
    id="loginPage"
  >
    <section
      id="login"
    >
      <center>
        <div
          className="login"
        >
          <h4>
            I see you are hungry once again, Please sign in
          </h4>
          <form
            id="loginForm"
            method="post"
            onSubmit={[Function]}
          >
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
            />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <button
              type="submit"
            >
              Sign in 
            </button>
          </form>
        </div>
      </center>
    </section>
    <div
      className="error hide"
    />
  </div>
</Login>
`;
